connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorMaker.java

```diff
@@ -322,6 +324,29 @@ public void run() {
         }
     }
 
+    /**
+     * Validates that the underlying connectors and tasks are running.
+     * Visible for testing purpose.
+     */
+    boolean isConfiguredAndRunning() {
+        return CONNECTOR_CLASSES.stream().allMatch(connectorClazz -> {
+            final String connName = connectorClazz.getSimpleName();
+            return herderPairs.stream().allMatch(sourceAndTarget -> {
+                final ConnectorStateInfo connectorStatus = this.connectorStatus(sourceAndTarget, connName);
+                return connectorStatus != null
+                    // verify that connector state is set to running
+                    && connectorStatus.connector().state().equals(AbstractStatus.State.RUNNING.toString())
+                    // verify that all tasks are set to running
+                    && connectorStatus.tasks().stream().allMatch(s -> s.state().equals(AbstractStatus.State.RUNNING.toString()));
```

Comment:
This doesn't actually verify that tasks are running (i.e., there could be an empty set of tasks for a connector and this condition would still be true).

We probably want to either 1) require that at least one task exists or 2) require the user to specify the expected number of tasks for each connector, in addition to this check (which ensures that every task that does exist is in the `RUNNING` state).